docs/ObserveApiProposal.md

```diff
@@ -0,0 +1,782 @@
+    exeuctionConfig {
+      instrument
+      ... on GmosNorthExecution {
+        static {
+          nodAndShuffle {
+            posA {
+              p { arcseconds }
+              q { arcseconds }
+            }
+            posB {
+              p { arcseconds }
+              q { arcseconds }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+Most instrument configurations will likely have the same structure, albeit with different individual
+static and dynamic (step) types.  For example,
+
+```
+type GmosNorthExecution implements ExecutionConfiguration {
+
+  instrument: InstrumentType!
+
+  # Static configuration that cannot change over the course of the sequence
+  static:      GmosNorthStatic!
+  
+  # Acquisition sequence, if any, is requested until the user confirms that the
+  # target is acquired.  Some calibration observations, for example, will have
+  # no acquisition steps.
+  aquisition:  ExecutionSequence!
+  
+  # Science sequence describes how to configure the instrument to acheive its
+  # science goals. 
+  science:     ExecutionSequence!  
+  
+}
+
+# The execution sequence pairs the next atom that should be executed with a 
+# `possibleFuture` sequence for reference only.  Once the `nextAtom` is
+# complete, a new query is required to get the new `nextAtom`.
+type ExecutionSequence {
+
+  # The next atom that should be executed, if any  
+  nextAtom:       GmosNorthAtom
+  
+  # Pontential future sequence, if any
+  possibleFuture: GmosNorthSequence
+  
+}
+  
+
+# -----------------------------------------------------------------------------
+# The types below are defined and to a large degree implemented.  Documentation
+# can be found in https://lucuma-odb-staging.herokuapp.com/playground.html
+#
+# This is simply an indication of what they contain.
+
+type GmosNorthStatic {
+
+  mosPreImaging: MosPreImaging!
+  
+  nodAndShuffle: NodAndShuffle
+  
+  stageMode:     GmosNorthStageMode!
+
+}
+
+type GmosNorthSequence {
+
+  # Atoms that make up the sequence, to be executed in order.
+  atoms: [GmosNorthAtom!]!
+
+  # Sum of all contained atoms' estimated times.
+  time:  StepTime!
+  
+}
+
+# Indivisible series of steps.  Breaking an atom implies losing the time
+# required to have obtained previous steps in the atom.  An example atom might
+# be a science frame and accompanying flat.  Without the flat, the science is
+# not usable.
+type GmosNorthAtom {
+
+  atomId: AtomId!
+  
+  # Sum of all step's estimated times.
+  time:  StepTime!
+  
+  # Steps that comprise the atom.  The must be executed in order.  
+  steps: [GmosNorthStep!]!
+
+}
+
+enum Breakpoint {
+
+  ENABLED
+  
+  DISABLED
+  
+}
+
+enum StepType {
+
+  BIAS
+  
+  DARK
+  
+  GCAL
+  
+  SCIENCE
+  
+  SMART_GCAL
+
+}
+
+type GmosNorthStep {
+
+  stepId:     StepId!
+
+  # Breakpoints that are programmed into the sequence generation, or present in
+  # manual sequence definitions.
+  breakpoint: Breakpoint!
```

Comment:
Is this a breakpoint that comes in the sequence definition? Should Observe stop the sequence before or after the step?
Are they related to the manual breakpoints set by the user while running the sequence in Observe? Can they be disabled/enabled by the Observe user?

Reply:
Yes, a breakpoint in the sequence definition.  The idea was that there are sometimes natural points in a sequence (as defined by instrument scientists) where the observer should stop and do something.  One such case is during acquisition to stop and see whether the target is acquired.  Whether there there other cases I'm not sure.

My thought was that it signaled a stop _before_ the step so that there is a way to signal a pause before beginning.

I think these would be the same as manual breakpoints set by the user.  How that works with "possible future" I'll admit to being unsure.  If the possible future changes then the breakpoints you may have carefully set could disappear.